law/target/collection.py

```diff
@@ -134,33 +138,51 @@ def exists(self, count=None, skip_optional=False):
         if count is not None:
             return count >= threshold
 
+        # helper to check for existence
+        def exists(t):
+            if skip_optional and t.optional:
+                return True
+            if isinstance(t, TargetCollection):
+                return t.exists(skip_optional=skip_optional)
+            return t.exists()
+
         # simple counting with early stopping criteria for both success and fail
         n = 0
         for i, targets in enumerate(self._iter_flat()):
-            if all(t.optional or t.exists() for t in targets):
+            if all(exists(t) for t in targets):
```

Comment:
This does not solve the issue I was trying to fix, because `exists()` is called without arguments and `skip_optional` defaults to `False`.

Reply:
But `skip_optional` is `True` as per

https://github.com/riga/law/blob/fix/optional_collections/law/task/base.py#L243-L246

no?